core/src/main/scala/com/yahoo/maha/core/query/QueryPipeline.scala

```diff
@@ -44,9 +44,20 @@ object QueryPipeline extends Logging {
   val asyncDisqualifyingSet: Set[Engine] = Set(DruidEngine)
 
   val completeRowList: Query => RowList = (q) => new CompleteRowList(q)
-  val dimDrivenPartialRowList: Query => RowList = (q) => new DimDrivenPartialRowList(q.queryContext.indexAliasOption.get, q)
-  val dimDrivenFactOrderedPartialRowList: Query => RowList = (q) => new DimDrivenFactOrderedPartialRowList(q.queryContext.indexAliasOption.get, q)
-  val factDrivenPartialRowList: Query => RowList = (q) => new FactDrivenPartialRowList(q.queryContext.indexAliasOption.get, q)
+  val dimDrivenPartialRowList: Query => RowList = (q) => {
+    val indexAlias = q.queryContext.indexAliasOption.get
+    val groupByKeys: List[String] = q.queryContext.factGroupByKeys diff List(indexAlias)
+    new DimDrivenPartialRowList(RowGrouping(q.queryContext.indexAliasOption.get, groupByKeys), q)
```

Comment:
This breaks the meaning of dim driven queries.  Dim driven queries will always have only 1 row for each key from driving dim.

Reply:
Thanks for the catch, that actually saved me some debugging.  :)